include/aklite.h

```diff
@@ -0,0 +1,178 @@
+      : name_(std::move(name)), sha256_(std::move(sha256)), version_(version), custom_(std::move(custom)) {}
+
+  /**
+   * Return the TUF Target name. This is the key in the targets.json key/value
+   * singed.metadata dictionary.
+   */
+  std::string Name() const { return name_; }
+  /**
+   * Return the sha256 OStree hash of the Target.
+   */
+  std::string Sha256Hash() const { return sha256_; }
+  /**
+   * Return the FoundriesFactory CI build number or in TUF, custom.version.
+   */
+  int Version() const { return version_; }
+
+  /**
+   * Return TUF custom data for a Target.
+   */
+  Json::Value Custom() const { return custom_; }
+
+ private:
+  std::string name_;
+  std::string sha256_;
+  int version_;
+  Json::Value custom_;
+};
+
+/**
+ * The response from an AkliteClient call to CheckIn()
+ */
+class CheckInResult {
+ public:
+  enum class Status {
+    Ok,        // check-in was good
+    OkCached,  // check-in failed, but locally cached meta-data is still valid
+    Failed,    // check-in failed and there's no valid local meta-data
+  };
+  CheckInResult(Status status) : status(status) {}
+  Status status;
+  std::vector<TufTarget> Targets() const { return targets_; }
+  TufTarget GetLatest() const { return targets_.back(); }
+
+ private:
+  friend class AkliteClient;
+  std::vector<TufTarget> targets_;
+};
+
+/**
+ * The response from an AkliteClient call to Download()
+ */
+class DownloadResult {
+ public:
+  enum class Status {
+    Ok,
+    DownloadFailed,
+    VerificationFailed,
+  };
+  DownloadResult(Status status, std::string description = "") : status(status), description(std::move(description)) {}
+  Status status;
+  std::string description;
+};
+
+/**
+ * The response from an AkliteClient call to Install
+ */
+class InstallResult {
+ public:
+  enum class Status {
+    Ok,
+    NeedsCompletion,
+    Failed,
+  };
+  InstallResult(Status status, std::string description = "") : status(status), description(std::move(description)) {}
+  Status status;
+  std::string description;
+};
+
+/**
+ * AkliteClient provides an easy-to-use API for users wanting to customize
+ * the behavior of aktualiz-lite.
+ */
+class AkliteClient {
+ public:
+  /**
+   * Construct a client instance pulling in config files from the given
+   * locations. ex:
+   *
+   *   AkliteClient c(AkliteClient::CONFIG_DIRS)
+   *
+   * @param config_dirs The list of files/directories to parse sota toml from.
+   */
+  AkliteClient(const std::vector<boost::filesystem::path> &config_dirs);
+  /**
+   * Used for unit-testing purposes.
+   */
+  AkliteClient(std::shared_ptr<LiteClient> client) : client_(client) {}
+
+  /**
+   * Performs a "check-in" with the device-gateay. A check-in consists of:
+   *  1) ask device-gateway for a new root.json - normally a 404.
+   *  2) ask device-gateway for timestamp and snapshot metadata.
+   *  3) pull down a new targets.json if needed
+   */
+  CheckInResult CheckIn(std::string hwid = "") const;
```

Comment:
What does `hwid` param is needed for? Isn't it defined in the one of toml files passed during the client instantiation? Or this is for the option that implies that each ECU has it's own target?

Reply:
At least in our use case, we'll probably always want to check for everything. It's slightly hard for me to imagine that ever not being the case, but I suppose it's plausible.

Reply:
One approach mike and I have thought for secondaries would be to embedded the secondary hash in the custom data of the primary's Target. Then you have atomic Targets and no need for this filter.

However, its probably worth the effort to allow it to filter for more than one hwid. As it stands now, its not good for either approach. I'll think about how to fix that.

Reply:
i've removed the hwid for now. I think when i do the next phase to support secondary ecu, we'll find a need to change this slightly.